---
title: ReferenceError
slug: Web/JavaScript/Reference/Global_Objects/ReferenceError
translation_of: Web/JavaScript/Reference/Global_Objects/ReferenceError
---
<div>{{JSRef("Global_Objects", "Error", "EvalError,InternalError,RangeError,ReferenceError,SyntaxError,TypeError,URIError")}}</div>

<h2 id="Summary">概述</h2>

<p><code><strong>ReferenceError（引用错误）</strong></code> 对象代表当一个不存在的变量被引用时发生的错误。</p>

<h2 id="Syntax">语法</h2>

<pre class="syntaxbox"><code>new ReferenceError([<var>message</var>[, <var>fileName</var>[, <var>lineNumber</var>]]])</code></pre>

<h3 id="Parameters">参数</h3>

<dl>
 <dt><code>message</code></dt>
 <dd>可选。描述可读的错误信息</dd>
 <dt><code>fileName</code> {{non-standard_inline}}</dt>
 <dd>可选。包含引起异常代码的文件名</dd>
 <dt><code>lineNumber</code> {{non-standard_inline}}</dt>
 <dd>可选。引起异常的代码行号</dd>
</dl>

<h2 id="Description">描述</h2>

<p>当你尝试引用一个未被定义的变量时，将会抛出一个 <code>ReferenceError</code> 。</p>

<h2 id="Properties">属性</h2>

<dl>
 <dt>{{jsxref("ReferenceError.prototype")}}</dt>
 <dd>Allows the addition of properties to an <code>ReferenceError</code> object.</dd>
</dl>

<h2 id="Methods">方法</h2>

<p>全局的 <code>ReferenceError</code> 本身并不包含有方法，但是他可以从原型链上继承一些方法</p>

<h2 id="ReferenceError_instances"><code>ReferenceError</code> 实例</h2>

<h3 id="Properties_of_ReferenceError_instances">属性</h3>

<div>{{page('/en-US/docs/Web/JavaScript/Reference/Global_Objects/ReferenceError/prototype', 'Properties')}}</div>

<h3 id="Methods_of_ReferenceError_instances">方法</h3>

<div>{{page('/en-US/docs/Web/JavaScript/Reference/Global_Objects/ReferenceError/prototype', 'Methods')}}</div>

<h2 id="Examples">例子</h2>

<h3 id="Example:_Catch_an_ReferenceError">例: 捕获一个 <code>ReferenceError</code></h3>

<pre class="brush: js">try {
  var a = undefinedVariable;
} catch (e) {
  console.log(e instanceof ReferenceError); // true
  console.log(e.message);                   // "undefinedVariable is not defined"
  console.log(e.name);                      // "ReferenceError"
  console.log(e.fileName);                  // "Scratchpad/1"
  console.log(e.lineNumber);                // 2
  console.log(e.columnNumber);              // 6
  console.log(e.stack);                     // "@Scratchpad/2:2:7\n"
}
</pre>

<h3 id="Example:_Create_an_ReferenceError">例: 创建一个 <code>ReferenceError</code></h3>

<pre class="brush: js">try {
  throw new ReferenceError('Hello', 'someFile.js', 10);
} catch (e) {
  console.log(e instanceof ReferenceError); // true
  console.log(e.message);                   // "Hello"
  console.log(e.name);                      // "ReferenceError"
  console.log(e.fileName);                  // "someFile.js"
  console.log(e.lineNumber);                // 10
  console.log(e.columnNumber);              // 0
  console.log(e.stack);                     // "@Scratchpad/2:2:9\n"
}
</pre>

<h2 id="规范">规范</h2>

<table class="standard-table">
 <tbody>
  <tr>
   <th scope="col">规范</th>
   <th scope="col">状态</th>
   <th scope="col">说明</th>
  </tr>
  <tr>
   <td>ECMAScript 3rd Edition.</td>
   <td>Standard</td>
   <td>Initial definition.</td>
  </tr>
  <tr>
   <td>{{SpecName('ES5.1', '#sec-15.11.6.3', 'ReferenceError')}}</td>
   <td>{{Spec2('ES5.1')}}</td>
   <td> </td>
  </tr>
  <tr>
   <td>{{SpecName('ES6', '#sec-native-error-types-used-in-this-standard-referenceerror', 'ReferenceError')}}</td>
   <td>{{Spec2('ES6')}}</td>
   <td> </td>
  </tr>
 </tbody>
</table>

<h2 id="浏览器兼容性">浏览器兼容性</h2>

{{Compat}}

<h2 id="See_also">相关链接</h2>

<ul>
 <li>{{jsxref("Error")}}</li>
 <li>{{jsxref("ReferenceError.prototype")}}</li>
</ul>
